boxes/boxes.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import logging
+import pathlib
+import requests
+import subprocess
+import tarfile
+import traceback
+from typing import Iterator, Tuple
+import urllib.parse
+
+
+def main() -> None:
+    args = parse_args()
+    if args.action == 'download':
+        Downloader(args.url, args.destination, args.extract).run()
+    else:
+        raise NotImplementedError()
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    actions = parser.add_subparsers(
+        title='action',
+        dest='action')
+
+    parser_download = actions.add_parser(
+        'download',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser_download.add_argument(
+        '-u', '--url',
+        help="Base boxes repository location",
+        metavar='URL',
+        type=str,
+        default='https://hitagi.dasie.mimuw.edu.pl/files/boxes/')
```

Comment:
Pewnie chcemy to ostatecznie mieć gdzieś na `downloads.sio2project.mimuw.edu.pl`, nie?

Reply:
Na razie zostawiłem TODO.

Reply:
ok

Reply:
ok